### PR TITLE
Arc 1622 api installtion

### PR DESCRIPTION
**Pull request by gxueatlassian**
**What's in this PR?**
Add gitHubAppId to the route and each handler.

**Why**
Just installationId is not enough.

**Added feature flags**
N/A

**Affected issues**  
ARC-1622

**How has this been tested?**  
Unit test.

**Whats Next?**
N/A

### CODE REVIEW

File: src/routes/api/installation/api-installation-delete.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { getJiraClient } from "~/src/jira/client/jira-client";
 
 export const ApiInstallationDelete = async (req: Request, res: Response): Promise<void> => {
 	const githubInstallationId = req.params.installationId;
```
Comment (rachellerathbone) [inline src/routes/api/installation/api-installation-delete.ts:6]: Nit: all casing for GH should be gitHub.

File: src/routes/api/installation/api-installation-router.ts [hunk 2]
```diff
@@ -7,25 +7,27 @@ import { ApiInstallationSyncPost } from "./api-installation-sync-post";
 import { ApiInstallationGet } from "./api-installation-get";
 
 export const ApiInstallationRouter = Router({ mergeParams: true });
+const subRouter = Router({ mergeParams: true });
+ApiInstallationRouter.use(`(/githubapp/:gitHubAppId(\\d+))?`, subRouter);
 
-ApiInstallationRouter.post(
+subRouter.post(
```
Comment (bgvozdev) [inline src/routes/api/installation/api-installation-router.ts:13]: Curious what would be the path to trigger resync for a cloud customer?

File: src/routes/api/installation/api-installation-router.ts [hunk 3]
```diff
@@ -7,25 +7,27 @@ import { ApiInstallationSyncPost } from "./api-installation-sync-post";
 import { ApiInstallationGet } from "./api-installation-get";
```
Comment (bgvozdev) [inline src/routes/api/installation/api-installation-router.ts:7]: I believe we also need to update all our runbooks?

File: src/routes/api/installation/api-installation-router.test.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
+import { Application } from "express";
```
Comment (bgvozdev) [inline src/routes/api/installation/api-installation-router.test.ts:1]: 🥇

File: src/routes/api/installation/api-installation-router.ts (hunk 3 above)
Comment (gxueatlassian) [reply src/routes/api/installation/api-installation-router.ts:7]: Ticket created ARC-1644.
For cloud, run book is same, so no changes.
For GHES, will do it in above ticket.

File: src/routes/api/installation/api-installation-router.ts (hunk 2 above)
Comment (gxueatlassian) [reply src/routes/api/installation/api-installation-router.ts:13]: Seems the resync is routed in the `ApiRouter` as
```javascript
ApiRouter.post(`/:uuid(${UUID_REGEX})?/resync`,..., async (req, res) => {});
```

So it is not included inside this ApiInstallationRouter.

File: src/routes/api/installation/api-installation-delete.ts (hunk 1 above)
Comment (gxueatlassian) [reply src/routes/api/installation/api-installation-delete.ts]: fixed.

File: src/routes/api/installation/api-installation-router.ts (hunk 3 above)
Comment (bgvozdev) [reply src/routes/api/installation/api-installation-router.ts:7]: Ok,sorry, I'm confused now :) Let me summarise my questions/concerns:

1. Could you please clarify who calls these endpoints?
2. Since we are prefixing it essentially with `githubapp` now, everything that was calling these endpoints will be broken now (no matter if it is in the code or in the runbooks), which is the cloud installations essentially (because GHES is not in prod yet)... What's the plan about it? Or noone is calling them now?

Comment (gxueatlassian) [reply src/routes/api/installation/api-installation-router.ts:7]: No worries, let me try to explain:
1. I believe only human or machine call these apis. (LOL, I mean pollinator or anyone does some ops in prod)
2. There's a `?` in the line `ApiInstallationRouter.use(`(/githubapp/:gitHubAppId(\\d+))?`, subRouter);`. I think that does the trick, that it can accept with or without the `/githubapp/:gitHubAppId` in the route. If you look closer to the test, it should test the scenarios that api is hitting both with and without the new route.

So in short, for current runbook (pollinator, human), cloud supporting is unchanged. Only added support is GHES, therefore I think runbook for GHES can be done in another ticket. 

Let me know if you spot anything wrong in the routing still. Thanks.

Comment (bgvozdev) [reply src/routes/api/installation/api-installation-router.ts:7]: Gotcha @gxueatlassian , thanks for the explanation! It makes much more sense now

**Review by bgvozdev — APPROVED** [review]
